File: leetcode-solutions/2831-number-of-beautiful-pairs/solution.py
import math
class Solution(object):
    def countBeautifulPairs(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        def gcd(a, b):
            while b != 0:
                a, b = b, a % b
            return a
        n = len(nums)
        count = 0

        for i in range (n):
            first = nums[i]
            while first >= 10:
                first //= 10
            for j in range (i + 1, n):
                last = nums[j] % 10
                if gcd(first, last) == 1:
                    count += 1

        return count

File: leetcode-solutions/2831-number-of-beautiful-pairs/solution.java
class Solution {
    public int countBeautifulPairs(int[] nums) {
        int n = nums.length;
        int count = 0;

        for(int i = 0;i < n;i++) {
            int first = nums[i];
            while(first >= 10)
                first /= 10;
            for(int j = i + 1;j < n;j++) {
                int last = nums[j] % 10;
                if(gcd(first, last) == 1) {
                    count++;
                }
            }
        }
        return count;
    }
    private int gcd(int a, int b) {
        while(b != 0) {
            int temp = b;
            b = a % b;
            a = temp;
        }
        return a;
    }
}
